### PR TITLE
DHFPROD-2009:DMSDK ingest/ FlowRunner fixes

### PR DESCRIPTION
**Pull request by srinathgit**
This PR contains
1. DMSDK ingestion 
2. Fixes to Flowrunner(writing job outputs to job doc, fixing an issue in jobs.sjs, add more info to flow status listener)

### CODE REVIEW

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/jobs.sjs [hunk 1]
```diff
@@ -189,7 +189,7 @@ class Jobs {
       docObj.batch.timeEnded = fn.currentDateTime();
     }
     if(error){
-      let stackTraceObj = error.stackTrace[0];
+      let stackTraceObj = error.stackFrames[0];
       docObj.batch.fileName = stackTraceObj.uri;
```
Comment (srinathgit) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/jobs.sjs]: @ryanjdew ,
'stackTrace' seems to be undefined in 'error' object. did you mean it to be 'stackFrames' as I see it has uri and line ?

File: marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/WriteStepRunner.java [hunk 2]
```diff
@@ -426,23 +478,102 @@ private Job runIngester(Job job) {
...
+            }
+            int i = 0;
+            while (i + 1 < perms.length) {
+                String roleName = perms[i++];
+                if (roleName == null || roleName.isEmpty()) {
+                    throw new IllegalArgumentException(
+                        "Illegal role name: " + roleName);
+                }
+                String perm = perms[i].trim();
+                if (!perm.equalsIgnoreCase(DocumentMetadataHandle.Capability.READ.toString()) &&
```
Comment (hao1st) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/WriteStepRunner.java]: we had better use if (!DocumentMetadataHandle.Capability.READ.toString().equalsIgnoreCase(perm)) ...

File: marklogic-data-hub/src/main/java/com/marklogic/hub/collector/impl/FileCollector.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        DiskQueue<String> results;
+        try {
+            results = new DiskQueue<>(10000);
+            Path dirPath = Paths.get(filePath);
+            if(! dirPath.isAbsolute()) {
+                File file = new File(hubConfig.getProjectDir(), dirPath.toString());
+                dirPath = file.toPath().toAbsolutePath();
+            }
+
+            if(! Files.exists(dirPath) && Files.isDirectory(dirPath)) {
```
Comment (hao1st) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/collector/impl/FileCollector.java]: Files.isDirectory(dirPath)  -> !Files.isDirectory(dirPath) ??

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/jobs.sjs (hunk 1 above)
Comment (ryanjdew) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/jobs.sjs]: @srinathgit Thanks for catching that.

**Pull request merged**